crates/jiter/src/py_lossless_float.rs

```diff
@@ -0,0 +1,71 @@
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
+use pyo3::types::PyType;
+
+use crate::Jiter;
+
+/// Represents a float from JSON, by holding the underlying bytes representing a float from JSON.
+#[derive(Debug, Clone)]
+#[pyclass]
+pub struct LosslessFloat(Vec<u8>);
+
+impl From<Vec<u8>> for LosslessFloat {
+    fn from(raw: Vec<u8>) -> Self {
+        Self(raw)
+    }
+}
+
+#[pymethods]
+impl LosslessFloat {
+    #[new]
+    fn new(raw: Vec<u8>) -> PyResult<Self> {
+        let s = Self(raw);
+        // check the string is valid by calling `as_float`
+        s.as_float()?;
+        Ok(s)
+    }
```

Comment:
It feels like there's a misalignment between the `From` implementation here without checking and the `new` implementation which is fallible. Should the `From` be a `TryFrom`? Should there be a `new_unchecked` if you need the infallible conversion? I think yes in both cases 😂

Reply:
changed to `new_unchecked`